app/models/chargeback_rate.rb

```diff
@@ -11,6 +20,8 @@ class ChargebackRate < ApplicationRecord
   validates_uniqueness_of   :guid
   validates_uniqueness_of   :description, :scope => :rate_type
 
+  virtual_column :assigned?, :type => :boolean
```

Comment:
> wow - never heard of run_callbacks. Would that set the errors variable that they UI can check?

The `errors.add` part in `def nope` will set the errors on the error object (the usual place the UI checks).

> Other question is your method nope ends in throw :abort. I have seen other samples on the web do return false instead. Which is the preferred way?

Ya, this is kind of an issue.  `return false` works on 4.2 but is deprecated in 5.0.  `throw :abort` is the preferred way on 5.0 but is not backwards compatible with 4.2 (IIRC).

> Final question is does ActiveRecord's destroy method raise errors that the caller needs to catch? Or is the destroyed? check adequate?

If you call `destroy!`, yes.  Calling `destroy` will return false, so if you ignore the return value then the failure is silently ignored.